src/decorations.ts

```diff
@@ -0,0 +1,74 @@
+import { Disposable, Event, workspace as vscodeWorkspace } from "vscode";
+import { PlasticScm } from "./plasticScm";
+import { PlasticSCMDecorationProvider } from "./decorationProvider";
+import { Workspace } from "./workspace";
+
+interface IDisposable {
+  dispose(): void;
+}
+
+function dispose<T extends IDisposable>(disposables: T[]): T[] {
+  disposables.forEach(d => d.dispose());
+  return [];
+}
+
+function filterEvent<T>(event: Event<T>, filter: (e: T) => boolean): Event<T> {
```

Comment:
Could we replace this with `filterEvent()` in `events.ts`? I believe they do the same thing